src/logic/note.ts

```diff
@@ -0,0 +1,206 @@
+import { useLiveQuery } from "dexie-react-hooks";
+import { CardType } from "./card";
+import { db } from "./db";
+import { v4 as uuidv4 } from "uuid";
+import { Deck } from "./deck";
+import { Table } from "dexie";
+
+export interface NoteSkeleton {
+  id: string;
+  deck: string;
+  referencedBy: string[];
+  creationDate: Date;
+  customOrder?: number;
+}
+
+export interface Note<T extends CardType> extends NoteSkeleton {
+  content: NoteContent<T>;
+}
+
+export function createNoteSkeleton(deck: string): NoteSkeleton {
+  return {
+    deck: deck,
```

Comment:
not really sure why you keep the deck ref in here...
I guess it's for easier access on learning phase or something? 

but you need to "join" to cards anyways no ?

Reply:
I found it pretty handy to access it directly in several cases, like in NoteMenu, everything with searching and filtering.

Actually, I would rather consider removing the deck property from cards... 😅